agents/acu/scan_helpers.py

```diff
@@ -195,6 +200,15 @@ def generate_linear_turnaround(az_endpoint1, az_endpoint2, az_speed, acc, el_end
         az_vel = -1*az_speed
     else:
         raise ValueError('Need two different motion endpoints')
+    if num_batches == None:
+        stop_iter = float('inf')
+    else:
+        stop_iter = num_batches
+        leftover_in_leg = (az_max - az_min) % daz
+        if leftover_in_leg == 0:
+            batch_size = (az_max - az_min) / daz
+        else:
+            batch_size = (az_max - az_min) / daz + 1
```

Comment:
This computation leaves batch_size as a float, and then range(batch_size) fails later.

If you are looking for a "round up" function, it is np.ceil (ceiling):
```
batch_size = int(np.ceil((az_max - az_min) / daz))
```

If I fix that, and pass num_batches=2, I get this trajectory:
![image](https://user-images.githubusercontent.com/1054988/164284687-66836d52-0008-47ef-bb90-bb7ecd996856.png)

Note how there is a single "dangling" point for the next scan.  Also, if you zoom in at t=0, you'll find there's no point there.

I think you should probably move your ```point_block[0].append(t + t0) ...``` lines from the end of each iteration to the start of each iteration.  That will fix the dangling point, and make sure you're actually starting from the starting point.

Reply:
This suggestion doesn't quite work (some cases had early dangling points instead of late ones). I think the new push fixes it though

Reply:
That's odd, I really thought ceil was right.

In the latest code, I'm getting dangling points in the case that daz divides (az_max - az_min) exactly:

![image](https://user-images.githubusercontent.com/1054988/164344559-3fc54ab0-56ea-4c5f-927c-2b7b9e58433e.png)

```
sh.generate_constant_velocity_scan(
        10, 1., 1., 1.2,
        30, 30, 0,
        step_time=0.05,
        start_time=0.,
        num_batches=2,
        ptstack_fmt=False)
```

Reply:
I think I figured out the issue. np.ceil looked incorrect because of the +1, but it is working with np.ceil and removing the +1